pkg/archive/archive.go

```diff
@@ -225,6 +225,60 @@ func CompressStream(dest io.Writer, compression Compression) (io.WriteCloser, er
 	}
 }
 
+// ReplaceFileTarWrapper converts inputTarStream to a new tar stream
+// with a single file called header.Name with contents of file.
+// TODO: make this into a generic tar conversion function with walkFn argument
```

Comment:
Also, I would rather avoid having two nested tar stream modifications in the trusted + stdin case.